crates/iota-indexer/src/lib.rs

```diff
@@ -230,16 +231,25 @@ impl Default for IotaNamesOptions {
 }
 
 pub async fn build_json_rpc_server(
+    store: PgIndexerStore,
     prometheus_registry: &Registry,
     reader: IndexerReader,
     config: &IndexerConfig,
     custom_runtime: Option<Handle>,
+    metrics: IndexerMetrics,
```

Comment:
This breaks the public API of the library.

Alternatively we can add a new `OptimisticWriteApi` that wraps `WriteApi` and holds the new optimistic methods.

Then we can add a new function here with the new signature to build the optimistic rpc-server.

Reply:
Okay, I'll create `build_optimistic_json_rpc_server` and `OptimisticWriteApi`

Reply:
6fed08b9353c8d77d66c3a7c3f476e99528c130f